components/TextFields/src/MDCTextInput.h

```diff
@@ -205,6 +205,22 @@ typedef NS_ENUM(NSUInteger, MDCTextInputTextInsetsMode) {
 
 @end
 
+@protocol MDCLeadingViewTextInput <MDCTextInput>
```

Comment:
We should put some docs for this protocol.

I assume we have a different one because we dont want to implement this for all of our textInput implementations. Once the old ones get deprecated is your long term plan to merge the protocols?

Reply:
Done.

This is just bringing out from inside the text field and making it possible to do introspection on it. That allows easier layout calculations in the controllers.

This is not implemented on multiline because multiline text fields should not have a leading view.